disnake/ui/button.py

```diff
@@ -41,15 +53,31 @@
 )
 
 if TYPE_CHECKING:
+    from typing_extensions import ParamSpec
+
     from ..emoji import Emoji
     from .item import ItemCallbackType
     from .view import View
 
+else:
+    ParamSpec = TypeVar
+
+T_co = TypeVar("T_co", covariant=True)
 B = TypeVar("B", bound="Button")
-V = TypeVar("V", bound="Optional[View]", covariant=True)
+B_co = TypeVar("B_co", bound="Button", covariant=True)
+V_co = TypeVar("V_co", bound="Optional[View]", covariant=True)
+P = ParamSpec("P")
 
 
-class Button(Item[V]):
+class Object(Protocol[T_co, P]):
```

Comment:
This is pretty smart, whoa
In any case, this could use a better name; and since it's duplicated in `select.py` it would be better to move this somewhere else, just `item.py` would be fine

Reply:
Yea, it's duplicated because I didn't necessarily want to modify any extra files. Will move it.
As for the naming, I'm dry on ideas. "Protocol which captures class initialization arguments". ObjectInitProto? Kinda meh

Reply:
I see, yea. Regarding naming, perhaps something like `ItemKind` since it's essentially a lite version of higher kinded typing, or just something simple like `ItemProto`... I can't really come up with any good ones either, it's just such a generic type/protocol haha

Reply:
I deliberately made the Object itself as generic as possible, since the protocol itself does not need any type constraints. Those are added later via `Obiect[B_co, ...]`